addons/hr/static/tests/m2x_avatar_employee.test.js

```diff
@@ -0,0 +1,541 @@
+
+    // click on first employee avatar
+    await contains(queryAll(".o_data_cell .o_m2o_avatar > img")[0]).click();
+    await contains(".o_avatar_card");
+    expect(".o_card_user_infos > span").toHaveText("Mario");
+    expect(".o_card_user_infos > a").toHaveText("Mario@partner.com");
+    expect(queryFirst(".o_avatar_card_buttons button").textContent).toBe("Send message");
+    await contains(queryFirst(".o_avatar_card_buttons button")).click();
+    await contains(".o-mail-ChatWindow");
+    expect(
+        queryOne(
+            ".o-mail-ChatWindow-header button.o-dropdown.o-mail-ChatWindow-command > .text-truncate"
+        ).innerText
+    ).toBe("Mario");
+
+    // click on second employee
+    await contains(queryAll(".o_data_cell .o_m2o_avatar > img")[1]).click();
+    expect(".o_card_user_infos span").toHaveText("Luigi");
+    expect(".o_avatar_card").toHaveCount(1);
+    expect(queryFirst(".o_avatar_card_buttons button").textContent).toBe("Send message");
+    await contains(queryFirst(".o_avatar_card_buttons button")).click();
+    expect(".o-mail-ChatWindow").toHaveCount(2);
+    expect(
+        queryAll(
+            ".o-mail-ChatWindow-header button.o-dropdown.o-mail-ChatWindow-command > .text-truncate"
+        )[0].textContent
+    ).toBe("Luigi");
+
+    // click on third employee (same as first)
+    await contains(queryAll(".o_data_cell .o_m2o_avatar > img")[2]).click();
+    expect(".o_card_user_infos span").toHaveText("Mario");
+    expect(".o_avatar_card").toHaveCount(1);
+    expect(queryOne(".o_card_user_infos span").textContent).toBe("Mario");
+    expect(".o_card_user_infos > a").toHaveText("Mario@partner.com");
+    expect(queryFirst(".o_avatar_card_buttons button").textContent).toBe("Send message");
+    await contains(queryFirst(".o_avatar_card_buttons button")).click();
+    expect(".o-mail-ChatWindow").toHaveCount(2);
+});
+
+test("many2one_avatar_employee widget in kanban view", async () => {
+    const { env } = await makeMockServer();
+    const partnerId = env["res.partner"].create({});
+    const userId = env["res.users"].create({ partner_id: partnerId });
+    const employeeId = env["hr.employee.public"].create({
+        user_id: userId,
+        user_partner_id: partnerId,
+    });
+    env["m2x.avatar.employee"].create({
+        employee_id: employeeId,
+        employee_ids: [employeeId],
+    });
+    onRpc("has_group", () => false);
+    await mountView({
+        type: "kanban",
+        resModel: "m2x.avatar.employee",
+        arch: `<kanban>
+            <templates>
+                <t t-name="card">
+                    <field name="employee_id" widget="many2one_avatar_employee"/>
+                </t>
+            </templates>
+        </kanban>`,
+    });
+    expect(queryFirst(".o_kanban_record").innerText.trim()).toBe("");
+    await contains(".o_m2o_avatar");
+    expect(queryAllAttributes(".o_m2o_avatar > img", "data-src")[0]).toBe(
```

Comment:
There's a `toHaveAttribute()`, better than `expect(query().something).toBe()` (shorter and better error message in case of failure)